integrationtest/helpers.go

```diff
@@ -390,6 +390,18 @@ func testK8sTokenTTL(t *testing.T, expectedSec int, token string) {
 	assert.Equal(t, expectedSec, int(exp-iat))
 }
 
+func testK8sTokenAudiences(t *testing.T, expectedAudiences []string, token string) {
+	parsed, err := josejwt.ParseSigned(token)
+	require.NoError(t, err)
+	claims := map[string]interface{}{}
+	err = parsed.UnsafeClaimsWithoutVerification(&claims)
+	require.NoError(t, err)
+	aud := claims["aud"].([]interface{})
+	for _, audience := range expectedAudiences {
+		assert.Contains(t, aud, interface{}(audience))
+	}
```

Comment:
I was playing around with this a bit, and noticed that if `expectedAudiences` is defined as a `[]interface{}` type, then we could do a simpler `assert.Equals()` here, which would also allow us to check that the default `aud` is what we expect (https://kubernetes.default.svc.cluster.local).

Reply:
I had tried `assert.Equal()` before, and the actual audiences seemed to have a different order each test run. I'm not sure the cause for it
```
--- FAIL: TestCreds_audiences (0.16s)
    --- FAIL: TestCreds_audiences/default_to_token_default_audiences (0.01s)
        helpers.go:400: 
            	Error Trace:	/Users/thy.ton/go/src/github.com/hashicorp/vault-plugin-secrets-kubernetes/integrationtest/helpers.go:400
            	            				/Users/thy.ton/go/src/github.com/hashicorp/vault-plugin-secrets-kubernetes/integrationtest/creds_integration_test.go:197
            	Error:      	Not equal: 
            	            	expected: []interface {}{"foo", "bar"}
            	            	actual  : []interface {}{"bar", "foo"}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,4 +1,4 @@
            	            	 ([]interface {}) (len=2) {
            	            	- (string) (len=3) "foo",
            	            	- (string) (len=3) "bar"
            	            	+ (string) (len=3) "bar",
            	            	+ (string) (len=3) "foo"
            	            	 }
            	Test:       	TestCreds_audiences/default_to_token_default_audiences
```

Reply:
`assert.ElementsMatch` might work then.